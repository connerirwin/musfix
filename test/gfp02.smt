(qualif Pos   ((v int)) (<= 0 v))
(qualif Neg   ((v int)) (<= v 0))
(qualif NeqZ  ((v int)) (not (= v 0)))
(qualif GtTen ((v int)) (< v 10))
(qualif False ()        (= 66 77))

(wf $k0 ((v0 int)))

(constraint
  (forall (v1 int)
          (=> ($k0 v1) (< 0 (+ v1 1)))))  ; 0 < (v1 + 1)
          
(constraint
  (forall (v3 int)
          (=> ($k0 v3) (> 11 (+ v3 1))))) ; 11 > (v3 + 1)

(constraint
  (forall (v2 int)
          (=> (= v2 9) ($k0 v2))))
          
(constraint
  (forall (v4 int)
        (=> (&& (< v4 9) (> v4 3)) ($k0 v4))))
