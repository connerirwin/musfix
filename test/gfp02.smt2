(qualif Pos   ((v Int)) (<= 0 v))
(qualif Neg   ((v Int)) (<= v 0))
(qualif NeqZ  ((v Int)) (not (= v 0)))
(qualif GtTen ((v Int)) (< v 10))
(qualif False ()        (= 66 77))

(wf $k0 ((v0 Int)))

(constraint
  (forall ((v1 Int))
          (=> ($k0 v1) (< 0 (+ v1 1)))))  ; 0 < (v1 + 1)

(constraint
  (forall ((v3 Int))
          (=> ($k0 v3) (> 11 (+ v3 1))))) ; 11 > (v3 + 1)

(constraint
  (forall ((v2 Int))
          (=> (= v2 9) ($k0 v2))))

(constraint
  (forall ((v4 Int))
        (=> (&& (< v4 9) (> v4 3)) ($k0 v4))))
