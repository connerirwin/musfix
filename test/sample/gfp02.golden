
Preparing to find greatest fixpoint...

Inputs
--------
Reading from file test/sample/gfp02.smt2
Qualifier "Pos" [Var IntS "v"] (Binary Le (IntLit 0) (Var IntS "v"))
Qualifier "Neg" [Var IntS "v"] (Binary Le (Var IntS "v") (IntLit 0))
Qualifier "NeqZ" [Var IntS "v"] (Unary Not (Binary Eq (Var IntS "v") (IntLit 0)))
Qualifier "GtTen" [Var IntS "v"] (Binary Lt (Var IntS "v") (IntLit 10))
Qualifier "False" [] (Binary Eq (IntLit 66) (IntLit 77))
WFConstraint "$k0" [Var IntS "v0"]
HornConstraint [Var IntS "v1"] (Binary Implies (Unknown (fromList [("v0",Var IntS "v1")]) "$k0") (Binary Lt (IntLit 0) (Binary Plus (Var IntS "v1") (IntLit 1))))
HornConstraint [Var IntS "v3"] (Binary Implies (Unknown (fromList [("v0",Var IntS "v3")]) "$k0") (Binary Gt (IntLit 11) (Binary Plus (Var IntS "v3") (IntLit 1))))
HornConstraint [Var IntS "v2"] (Binary Implies (Binary Eq (Var IntS "v2") (IntLit 9)) (Unknown (fromList [("v0",Var IntS "v2")]) "$k0"))
HornConstraint [Var IntS "v4"] (Binary Implies (Binary And (Binary Lt (Var IntS "v4") (IntLit 9)) (Binary Gt (Var IntS "v4") (IntLit 3))) (Unknown (fromList [("v0",Var IntS "v4")]) "$k0"))

QMAP
--------
fromList [("$k0",QSpace {_qualifiers = [Binary Le (IntLit 0) (Var IntS "v0"),Binary Le (Var IntS "v0") (IntLit 0),Unary Not (Binary Eq (Var IntS "v0") (IntLit 0)),Binary Lt (Var IntS "v0") (IntLit 10),Binary Eq (IntLit 66) (IntLit 77)], _maxCount = 5})]

Candidates
--------
[Candidate {solution = fromList [("$k0",fromList [Binary Lt (Var IntS "v0") (IntLit 10),Binary Le (IntLit 0) (Var IntS "v0")])], validConstraints = fromList [Binary Implies (Unknown (fromList [("v0",Var IntS "v1")]) "$k0") (Binary Lt (IntLit 0) (Binary Plus (Var IntS "v1") (IntLit 1))),Binary Implies (Unknown (fromList [("v0",Var IntS "v3")]) "$k0") (Binary Gt (IntLit 11) (Binary Plus (Var IntS "v3") (IntLit 1))),Binary Implies (Binary Eq (Var IntS "v2") (IntLit 9)) (Unknown (fromList [("v0",Var IntS "v2")]) "$k0"),Binary Implies (Binary And (Binary Lt (Var IntS "v4") (IntLit 9)) (Binary Gt (Var IntS "v4") (IntLit 3))) (Unknown (fromList [("v0",Var IntS "v4")]) "$k0")], invalidConstraints = fromList [], label = "0"}]


Final candidates: 
0: [$k0 -> {v0 [37m<[0m [36m10[0m, [36m0[0m [37m<=[0m v0}] (4 0)

